src/states/common/approved.rs

```diff
@@ -312,6 +312,16 @@ pub trait Approved: Base {
             our_pfx = *self.chain_mut().our_prefix();
         }
 
+        // Checking members vote status and vote Offline for non-resposive nodes.
```

Comment:
I think that make sense, here, i.e when we accumulate a new event is where we have the case where a new node may be detected unresponsive. Could you move it to a well named sub function though.